lib/sprockets/exporting.rb

```diff
@@ -0,0 +1,56 @@
+module Sprockets
+  # `Exporting` is an internal mixin whose public methods are exposed on
+  # the `Environment` and `CachedEnvironment` classes.
+  module Exporting
+    # Exporters are ran on the assets:precompile task
+    def exporters
+      config[:exporters]
+    end
+
+    # Registers a new Exporter `klass` for `mime_type`.
+    #
+    #     register_preprocessor '*/*', Sprockets::GzipExporter
+    #
+    def register_exporter(mime_types, proc = nil, &block)
+      proc ||= block
+
+      if mime_types.is_a? String
+        mime_types = [mime_types]
+      end
+      mime_types.each do |mime_type|
+        self.config = hash_reassoc(config, :exporters, mime_type) do |_exporters|
+          _exporters << proc
+        end
+      end
+    end
+
+    def unregister_exporter(mime_types, exporter = nil)
+      unless mime_types.is_a? Array
+        if mime_types.is_a? String
+          mime_types = [mime_types]
+        elsif mime_types < Exporter
```

Comment:
It evaluates to `true` if `mime_types` is a class that inherits from `Exporter` (which is different from `is_a?` because `mime_types` is a class, not an instance)